docs/prometheus-metrics.md

```diff
@@ -0,0 +1,58 @@
+# Prometheus Metrics Coverage
+
+We plan to collect a rich set of metrics in kubeflow/common's `JobController` using [Prometheus](https://prometheus.io/).
+The goal is to report generic metrics (e.g. metrics related to pods/jobs/services) during the lifecycle of `JobController` so that:
+
+* Other operators built on top of it will automatically report Prometheus metrics without additional efforts;
+* It is easier for users of Kubeflow distributed training operators to monitor operator performance and behaviors using consistent set of metrics for different distributed training operators.
+
+This document outlines the list of Prometheus metrics we plan to cover in `JobController`.
+
+## Pod Metrics
+
+The following metrics related to the lifecycle of pods will be reported:
+
+* The total number of created pods
+* The total number of restarted pods
+* The total number of deleted pods
+* The total number of failed pods
+
+The following metrics will be reported on each pod:
+
+* CPU usage
+* GPU usage
+* Memory usage
+* Network usage
+* I/O usage
+* Keep-Alive check
+* Is-leader check
```

Comment:
the leader means training job leader?

Reply:
I renamed it and added description. Please let me know if it's still unclear.